go/felix/intdataplane/endpoint_mgr.go

```diff
@@ -56,32 +57,48 @@ type endpointManager struct {
 	routeTable   routeTable
 	writeProcSys procSysWriter
 
+	// Pending updates, cleared in CompleteDeferredWork as the data is copied to the activeXYZ
+	// fields.
+	pendingWlEpUpdates  map[proto.WorkloadEndpointID]*proto.WorkloadEndpoint
+	pendingIfaceUpdates map[string]ifacemonitor.State
+
 	// Active state, updated in CompleteDeferredWork.
-	activeEndpoints      map[proto.WorkloadEndpointID]*proto.WorkloadEndpoint
-	activeIfaceNameToID  map[string]proto.WorkloadEndpointID
-	activeUpIfaces       set.Set
-	activeIdToChains     map[proto.WorkloadEndpointID][]*iptables.Chain
-	activeDispatchChains []*iptables.Chain
-
-	ifaceNamesToReconfigure set.Set
-	ifaceIDsToUpdateStatus  set.Set
-
-	// Pending updates, cleared in CompleteDeferredWork.
-	pendingEndpointUpdates map[proto.WorkloadEndpointID]*proto.WorkloadEndpoint
-	pendingIfaceUpdates    map[string]ifacemonitor.State
-
-	// Host endpoint processing.
-	ifaceAddrs               map[string]set.Set
-	rawHostEndpoints         map[proto.HostEndpointID]*proto.HostEndpoint
-	dirtyHostEndpoints       bool
-	activeHostIfaceToChains  map[string][]*iptables.Chain
+	activeWlEndpoints     map[proto.WorkloadEndpointID]*proto.WorkloadEndpoint
+	activeWlIfaceNameToID map[string]proto.WorkloadEndpointID
+	activeUpIfaces        set.Set
+	activeWlIDToChains    map[proto.WorkloadEndpointID][]*iptables.Chain
+	activeDispatchChains  []*iptables.Chain
+
+	// wlIfaceNamesToReconfigure contains names of workload interfaces that need to have
+	// their configuration (sysctls etc.) refreshed.
+	wlIfaceNamesToReconfigure set.Set
+
+	// epIDsToUpdateStatus contains IDs of endpoints that we need to report status for.
+	// Mix of host and workload endpoint IDs.
+	epIDsToUpdateStatus set.Set
+
+	// hostIfaceToAddrs maps host interface name to the set of IPs on that interface (reported
+	// fro the dataplane).
+	hostIfaceToAddrs map[string]set.Set
+	// rawHostEndpoints contains the raw (i.e. not resolved to interface) host endpoints.
+	rawHostEndpoints map[proto.HostEndpointID]*proto.HostEndpoint
+	// hostEndpointsDirty is set to true when host endpoints are updated.
+	hostEndpointsDirty bool
+	// activeHostIfaceToChains maps host interface name to the chains that we've programmed.
+	activeHostIfaceToChains map[string][]*iptables.Chain
+	// activeHostDispatchChains contains the dispatch chains that we've programmed for host
+	// endpoints.
 	activeHostDispatchChains []*iptables.Chain
+	// activeHostEpIDToIfaceNames records which interfaces we resolved each host endpoint to.
```

Comment:
But it's an error, isn't it?  (I think it's only possible if the same IP address is assigned to multiple interfaces.)